api/server/sdk/cluster_domains.go

```diff
@@ -0,0 +1,80 @@
+package sdk
+
+import (
+	"context"
+
+	"github.com/libopenstorage/openstorage/api"
+	"github.com/libopenstorage/openstorage/cluster"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+// ClusterDomainsServer is an implementation of the gRPC OpenStorageClusterDomains interface
+type ClusterDomainsServer struct {
+	server serverAccessor
+}
+
+func (m *ClusterDomainsServer) cluster() cluster.Cluster {
+	return m.server.cluster()
+}
+
+func (m *ClusterDomainsServer) Enumerate(
```

Comment:
Use the function in the `api.IsAdmin(ctx) bool` (when available) to make sure that these functions are only conducted by the storage system administrator

Reply:
#977 adds this support, as `api.IsAdminByContext(ctx)`

Reply:
done